enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,329 @@
+
+```markdown
+playbooks/
+  example-e2e-workflow.yml
+roles/
+    common/
+        tasks/
+        handlers/
+        library/
+        files/
+        templates/
+        vars/
+        defaults/
+        meta/
+    example_role/
+        tasks/
+        defaults/
+        meta/
+```
+
+This is pretty open ended and meant to be. I just believe that focusing on making
+reusable roles here for the steps is key.
+
+These are my thoughts about Openshift CI and the step registry and making it easy to understand. 
+Copy the `pre`, `test`, and `post` pattern locally in the playbooks and include the playbooks like so:
+
+```markdown
+# File structure:
+playbooks/
+  example-e2e-workflow.yml
+  example-e2e-workflow_pre.yml
+  example-e2e-workflow_test.yml
+  example-e2e-workflow_post.yml
+
+# example-e2e-workflow.yml
+- hosts: localhost
+
+- name: Run e2e-workflow Pre steps
+  import_playbook: example-e2e-workflow_pre.yml
+
+- name: Run e2e-workflow Test steps
+  import_playbook: example-e2e-workflow_test.yml
+
+- name: Run e2e-workflow Post steps
+  import_playbook: example-e2e-workflow_test.yml
+```
+
+This is just a general idea and would like to keep this pretty open ended at the moment.
+The first priority will be to run these playbooks locally but how to understand this in the CI
+is being taken into consideration.
+
+Another thing that was brought up is that in Openshift CI the way we pass parameters is
+through the use of environment variables. Which means all of these Ansible roles will
+need to have defaults and be aware of those environment variables.
+
+#### Build
+
+The scripts/automation for building the agent-based installer should be inside the repository
+where the installer is being worked on and not dev-scripts. We will reference the built 
+container images and ISO in dev-scripts.
+
+#### Deployment
+
+Deploying the agent-based installer will be heavily linked to the environment set up. This
+logic will land in dev-scripts alongside the pre environment setup steps.
+
+#### Tests
+
+For the actual test scenarios themselves we should use Ginkgo/Gomega. This is 
+similar to the rest of Openshift. Well most of (if not all) the product is written in Golang
+so it makes sense to use a Golang framework for these.
+
+There is a repository created as a preliminary place to put these. Plan to use a similar
+structure to openshift-tests in this repository.
+
+We will be including existing e2e tests that have been created to validate the OCP cluster
+installed by the new agent-based installer.
+
+#### Teardown
+
+Currently there is a `make clean` flow for starting new locally. We will need to add similar
+functionally for the flows that will be used for the agent-based installer
+
+### Open Questions [optional]
+
+### Test Plan
+
+* e2e tests for the three install scenarios of cluster 0 (SNO, Compact, HA)
+* Run Openshift compliance tests against these installed clusters
```

Comment:
If 2 teams will be using dev-scripts for slightly different purposes, we will want test jobs to run on changes in the dev-scripts repo to ensure neither team accidentally breaks the workflow of the other.

Reply:
Agreed. To be clear this is now 3 teams too :) So this risk is something to keep an eye on.